grid-demo/src/app/angular/advanced-row-selection/custom-clr-advanced-selection.directive.ts

```diff
@@ -13,33 +13,27 @@ export class CustomClrAdvancedSelectionDirective implements OnInit {
   }
 
   @HostListener('click', ['$event'])
-  overrideSelection(evt: MouseEvent) {
-    const target = evt.target as HTMLElement;
+  overrideSelection(event: MouseEvent) {
+    const target = event.target as HTMLElement;
 
     if (
-      !isSpecialKeyClick(evt) &&
-      !isCheckboxClick(target) &&
-      hasDatagridRowParent(target, this.elementRef.nativeElement)
+      !mouseEventIsSpecialKeyClick(event) &&
+      !elementHasSelectionCellParent(target, this.elementRef.nativeElement) &&
+      elementHasDatagridRowParent(target, this.elementRef.nativeElement)
     ) {
       this.datagrid.selected = [];
     }
   }
 }
 
-function isSpecialKeyClick(evt: MouseEvent) {
-  return evt.shiftKey || evt.ctrlKey || evt.metaKey;
+function mouseEventIsSpecialKeyClick(event: MouseEvent) {
+  return event.shiftKey || event.ctrlKey || event.metaKey;
 }
 
-function isCheckboxClick(target: HTMLElement | null) {
-  return target?.parentElement?.className === 'clr-checkbox-wrapper';
+function elementHasSelectionCellParent(element: HTMLElement | null, maxParent: HTMLElement) {
+  return maxParent.contains(element?.closest('.datagrid-select') || null);
```

Comment:
What is a selection cell parent?

Reply:
It's the entire cell that contains the selection checkbox.